File: game_app/event_handler_generics.py
class generic_event_handler:
    key_to_handler_dict = {}
    def register_handler(self,key,handler):
        self.key_to_handler_dict[key] = handler;

class value_is_key_event_handler(generic_event_handler):
    def __init__(self,key_key,payload_key):
        self.key_key = key_key
        self.payload_key = payload_key
    def act_on (self,data_dict,*args):
        key = data_dict[self.key_key]
        payload = data_dict[self.payload_key]
        if key in self.key_to_handler_dict:
            handler = self.key_to_handler_dict[key]
            handler(payload,*args);

class key_is_key_event_handler(generic_event_handler):
    def act_on (self,data_dict,*args):
        for key in data_dict:
            if key in self.key_to_handler_dict:
                payload = data_dict[key]
                handler = self.key_to_handler_dict[key]
                handler(payload,*args);

File: game_app/static/homemade/js/event_handler_generics.js
function generic_event_handler(){
	var key_to_handler_dict = {};
	this.register_handler = function (key,handler){
		key_to_handler_dict[key] = handler;
	}
}

function value_is_key_event_handler(key_key,payload_key,context){
	var key_to_handler_dict = {};
	this.register_handler = function (key,handler, context){
		key_to_handler_dict[key] = [handler, context];
	}
	this.act_on = function(dict){
		var key = dict[key_key];
		var payload = dict[payload_key];
		if (key_to_handler_dict.hasOwnProperty(key)) {
			var handler_info = key_to_handler_dict[key];
			handler = handler_info[0];
			context = handler_info[1];
			handler.apply(context, [payload]);
        }
    }
}

function key_is_key_event_handler(){
	var key_to_handler_dict = {};
	this.register_handler = function (key,handler,context){
		key_to_handler_dict[key] = [handler, context];
	}
	this.act_on = function(dict){
		for (var key in dict) {
			var payload = dict[key];
	        if (key_to_handler_dict.hasOwnProperty(key)) {
	        	var handler_info = key_to_handler_dict[key];
	        	handler = handler_info[0];
	        	context = handler_info[1];
	        	handler.apply(context, [payload]);
	        }
		}
    }
}